client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -144,16 +145,16 @@ public Response performRequest(String method, String endpoint, Header... headers
      * @throws ClientProtocolException in case of an http protocol error
      * @throws ResponseException in case elasticsearch responded with a status code that indicated an error
      */
-    public Response performRequest(String method, String endpoint, Map<String, String> params, Header... headers) throws IOException {
-        return performRequest(method, endpoint, params, null, headers);
+    public Response performRequest(String method, String endpoint, Map<String, String> params, Header... headers) throws Exception {
+        return performRequest(method, endpoint, params, (HttpEntity)null, headers);
     }
 
     /**
-     * Sends a request to the elasticsearch cluster that the current client points to.
-     * Selects a host out of the provided ones in a round-robin fashion. Failing hosts are marked dead and retried after a certain
-     * amount of time (minimum 1 minute, maximum 30 minutes), depending on how many times they previously failed (the more failures,
-     * the later they will be retried). In case of failures all of the alive nodes (or dead nodes that deserve a retry) are retried
-     * till one responds or none of them does, in which case an {@link IOException} will be thrown.
+     * Sends a request to the elasticsearch cluster that the current client points to. Blocks till the request is completed and returns
```

Comment:
"till" is funny because it used to be common in English then became uncommon and then became slang. You'd see it in [shakespeare](http://nfs.sparknotes.com/hamlet/page_60.html). So if you want to sound super fancy you can say it with other old school words. Now it is also slang for "until". Also it could be a noun: a cash register. Or a verb: to break apart land so you can put seeds in it. English is horrible.